modules/component/src/core/cd-aware/nameToStrategy.ts

```diff
@@ -0,0 +1,22 @@
+import { Observable } from 'rxjs';
+import { distinctUntilChanged, map, startWith } from 'rxjs/operators';
+import {
+  CdStrategy,
+  DEFAULT_STRATEGY_NAME,
+  StrategySelection,
+} from './strategy';
+
+export function nameToStrategy<U>(strategies: StrategySelection<U>) {
+  return (o$: Observable<string>): Observable<CdStrategy<U>> => {
+    return o$.pipe(
+      distinctUntilChanged(),
+      startWith(DEFAULT_STRATEGY_NAME),
+      map(
+        (strategy: string): CdStrategy<U> =>
+          strategies[strategy]
+            ? strategies[strategy]
+            : strategies[DEFAULT_STRATEGY_NAME]
```

Comment:
Reads a bit easier imho

```suggestion
          strategies[strategy] || strategies[DEFAULT_STRATEGY_NAME]
```

Reply:
Very nice!